crates/nostr-sdk/src/client/mod.rs

```diff
@@ -1101,17 +1101,56 @@ impl Client {
         self.send_event_to(urls, event).await
     }
 
-    /// Get public key metadata
+    /// Fetch public key metadata from database, falling back to connected relays.
     ///
     /// <https://github.com/nostr-protocol/nips/blob/master/01.md>
+    pub async fn fetch_metadata(
+        &self,
+        public_key: PublicKey,
+        timeout: Option<Duration>,
+    ) -> Result<Metadata, Error> {
+        let filter: Filter = Filter::new()
+            .author(public_key)
+            .kind(Kind::Metadata)
+            .limit(1);
+        let events: Vec<Event> = self
+            .get_events_of(vec![filter], EventSource::both(timeout))
+            .await?;
+        match events.first() {
+            Some(event) => Ok(Metadata::try_from(event)?),
+            None => Err(Error::MetadataNotFound),
+        }
+    }
+
+    /// Get cached public key metadata.
+    ///
+    /// <https://github.com/nostr-protocol/nips/blob/master/01.md>
+    pub async fn get_metadata(&self, public_key: PublicKey) -> Result<Metadata, Error> {
```

Comment:
The `NostrDatabase` already have a method for this. So I guess could be specified in the `fetch_metadata`  docs and we can remove `get_metadata`?

```rust
let database = client.database();
let profile = database.profile(public_key).await?;
println!("{}", profile.metadata.as_json());
```